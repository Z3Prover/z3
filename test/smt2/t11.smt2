
; Copyright (c) 2015 Microsoft Corporation


(declare-fun f (Int Int) Int)

(set-option :pp.max-depth 10)

(display (forall ((x Int) (y0 Int))
                 (let ((y (f x y0)))
                   (or (forall ((z Int))
                            (or (= (f z x) (f y x))
                                       (exists ((w Int))
                                               (= (f z w) (f y x)))))
                       (> (f y y) 0)))))
        