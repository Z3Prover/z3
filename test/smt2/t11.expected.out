(forall ((x Int) (y0 Int))
  (or (forall ((z Int))
        (or (= (f z x) (f (f x y0) x))
            (exists ((w Int)) (= (f z w) (f (f x y0) x)))))
      (> (f (f x y0) (f x y0)) 0)))
